lib/inspec/profile.rb

```diff
@@ -231,17 +230,33 @@ def collect_tests(include_list = @controls)
 
           ## Pull together waiver
           if waiver_path
-            waived_controls = YAML.load(waiver_path).keys
+            waived_controls = YAML.load_file(waiver_path).keys
           end
           regex_matcher = "(#{waived_controls.join('|')})"
 
           ## Purge tests (this could be doone in next block for performance)
+          ## TODO: implement earlier with pure AST and pure autocorrect AST
           purged_tests = {}
-          tests.each do |key, value|
-            cleared_tests = value.split("control ").select do |element|
-              !element&.match?(regex_matcher)
-            end.join("control ")
-            purged_tests[key] = cleared_tests
+          if Inspec::Config.cached["retain_waiver_data"]
+            # VERY EXPERIMENTAL, but an empty describe block at the top level
+            # of the control blocks evaluation of ruby code until later-term
+            # waivers. This allows current waiver-data (e.g. skipis) to still
+            # be processed and rendered
+            tests.each do |key, value|
+              cleared_tests = value.split("control ").collect do |element|
+                next if element.blank?
+                splitlines = element.split("\n")
+                new_element = splitlines[0] + "\ndescribe '---' do\n" + splitlines[1..-1].join("\n") + "\nend\n"
+              end.join("control ")
+              purged_tests[key] = cleared_tests
+            end
```

Comment:
This needs to be looked at very closely, this seems.... fishy.

Reply:
If there's any way someone can find the lazy evaluation tactics used by RSpec in 'describe' and safely port that approach to 'control' we'd be in business.

I lost a few hours in a rabbit hole on that one.

Nothing inside a 'describe' block is evaluated when a controller is waived. (It can be tested with a 'raise' or 'sleep'). So until 'control' can be modified this would wrap all code in an unnamed 'describe' block at runtime.

I concur that the current approach is fishy- but should work in a desperate situation.

I hope I can find a few hours soon just to look at 'control' and the guts of 'describe' again.

If anyone cracks the 'control' case first let me know as I'd be very interested how it was done!